lib/internal/util/comparisons.js

```diff
@@ -249,6 +250,10 @@ function innerDeepEqual(val1, val2, strict, memos) {
              isNativeError(val2) ||
              val2 instanceof Error) {
     return false;
+  } else if (isURL(val1)) {
+    if (!isURL(val2) || val1.href !== val2.href) {
+      return false;
+    }
```

Comment:
```suggestion
    return isURL(val2) && val1.href === val2.href
```

Reply:
This will break the case like:
```javascript
const a = new URL('http://foo');
const b = new URL('http://foo');
a.bar = 1;
b.bar = 2;
assertNotDeepOrStrict(a, b);
```